### PR TITLE
3 new arm tool set implants

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request

Adds in 3 count them 3 new implants!
Surgery tools now have a knife locked behind an emag
Cleans up the implant file a bit removing some unneeded subtypes and makes the file look nice
Shield Arm implant, a shield implant that can not be dropped yes, but blocks less melee then normal.
Best use against ranged based weapons as that gives more time to react(aka block) then being up close. - Emag is said to unlock an advanced  flash
Janitor Toolset - Has a mop and everything a janitor could ever want or need in their bag other then tools and floor replacement, usefull to save space on items you have to carry. - Emag is said to unlock EVIL soap and some slippery spray
Service Toolset - This mismash of items is for cook/barkeep, to help them cook, mix and tend to the people that wish to get food/wasted! - Emag is said to unlock a knife!

## Why It's Good For The Game

These implants are really just to help out later one and give implants/medical more things to do well helping them save space for other items they need for their job. This also gives way to maybe more implants regarding a job, like maybe some more traitor/ninja implants! - More for later

## Changelog
:cl:
add: Three new arm implants, shield for sec, janitor and service
/:cl:

### CODE REVIEW

File: code/game/objects/items/shields.dm [hunk 1]
```diff
@@ -236,3 +236,20 @@
...
+	icon = 'icons/obj/items_and_weapons.dmi'
+	block_chance = 30 //May be big but hard to move around to block.
+	force = 10
+	slowdown = 1
+	transparent = FALSE
+	item_flags = SLOWS_WHILE_IN_HAND
+
+/obj/item/shield/riot/implant/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(attack_type == PROJECTILE_ATTACK)
+		final_block_chance = 60 //Massive shield
```
Comment (silicons) [inline code/game/objects/items/shields.dm:254]: Can we not?

Comment (Trilbyspaceclone) [reply code/game/objects/items/shields.dm:254]: ?
its... 10% more then a normal shield, as well as FINAL BLOCK always being 60% rather then buffing to be 70% with drink... 
Whats the problem

Comment (necromanceranne) [reply code/game/objects/items/shields.dm:254]: Actually, for something that requires surgery and eats up an arm slot for the much more useful utility of the toolset arm, it's probably not bad to have it possess a distinct advantage over standard equipment. The toolspeeds of the implant tools is really high if I remember correctly. 10% doesn't realistically make THAT much of a difference, but it is a no-drop shield.

Also, do people really abuse Alexander that much? It's alcohol content is fairly high, you'd need a cyberliver to really make the most of it.

Comment (Trilbyspaceclone) [reply code/game/objects/items/shields.dm:254]: The problem with Alexander drink is that you can abuse it a lot. And 70% projectile block is to much for me to even.
40% block with the drink is ok in my eyes as its still less then normal shields
As for it being better then normal gear, it has a slowdown and less damage so that getting the combo if no drop + shield is better then the implant.
This is not ment to be the `Oh shit install this into my body` every round

File: code/modules/surgery/organs/augments_arms.dm [hunk 2]
```diff
@@ -193,16 +164,76 @@
...
+	return TRUE
 
+/obj/item/organ/cyberimp/arm/janitor
+	name = "janitor toolset implant"
+	desc = "A set of surgical tools hidden behind a concealed panel on the user's arm."
+	contents = newlist(/obj/item/lightreplacer, /obj/item/holosign_creator, /obj/item/soap/nanotrasen, /obj/item/reagent_containers/spray/cyborg_drying, /obj/item/mop/advanced, /obj/item/paint/paint_remover, /obj/item/reagent_containers/glass/beaker/large, /obj/item/reagent_containers/spray/cleaner) //Beaker if for refilling sprays
+
+/obj/item/organ/cyberimp/arm/janitor/emag_act()
+	. = ..()
+	if(locate(/obj/item/soap/syndie) in items_list) //If we have syndie soap we have the other items
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: Check for the emagged object flag instead and add an `obj_flags |= EMAGGED` line two rows below.

File: code/modules/surgery/organs/augments_arms.dm [hunk 3]
```diff
@@ -193,16 +164,76 @@
...
+	return TRUE
+
+/obj/item/organ/cyberimp/arm/service
+	name = "service toolset implant"
+	desc = "A set of tools needed for any barkeep or cook on the user's arm."
+	contents = newlist(/obj/item/extinguisher/mini, /obj/item/kitchen/knife/combat/bone/plastic, /obj/item/hand_labeler, /obj/item/pen, /obj/item/reagent_containers/dropper, /obj/item/kitchen/rollingpin, /obj/item/reagent_containers/glass/beaker/large, /obj/item/reagent_containers/syringe,/obj/item/reagent_containers/food/drinks/shaker, /obj/item/radio/off, /obj/item/camera, /obj/item/modular_computer/tablet/preset/cargo)
+
+/obj/item/organ/cyberimp/arm/service/emag_act()
+	. = ..()
+	if(locate(/obj/item/kitchen/knife/combat/cyborg) in items_list)
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: Idem.

File: code/modules/surgery/organs/augments_arms.dm [hunk 4]
```diff
@@ -231,7 +262,26 @@
...
+	contents = newlist(/obj/item/melee/transforming/energy/blade/hardlight)
+
+/obj/item/organ/cyberimp/arm/shield
+	name = "arm-mounted riot shield"
+	desc = "A deployable riot shield to help deal with civil unrest."
+	contents = newlist(/obj/item/shield/riot/implant)
+
+/obj/item/organ/cyberimp/arm/shield/emag_act()
+	. = ..()
+	if(locate(/obj/item/assembly/flash/armimplant) in items_list)
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: Idem.

File: code/modules/surgery/organs/augments_arms.dm [hunk 5]
```diff
@@ -231,7 +262,26 @@
...
+		return
+	to_chat(usr, "<span class='notice'>You unlock [src]'s integrated knife!</span>")
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: It's not really a knife.

File: code/modules/surgery/organs/augments_arms.dm [hunk 6]
```diff
@@ -193,16 +164,76 @@
...
+	. = ..()
+	if(locate(/obj/item/kitchen/knife/combat/cyborg) in items_list)
+		return
+	to_chat(usr, "<span class='notice'>You unlock [src]'s integrated knife!</span>")
+	items_list += new /obj/item/kitchen/knife/combat/cyborg(src)
+	return TRUE
 
+/obj/item/organ/cyberimp/arm/janitor
+	name = "janitor toolset implant"
+	desc = "A set of surgical tools hidden behind a concealed panel on the user's arm."
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: "janitorial tools" is what you mean.

File: code/modules/surgery/organs/augments_arms.dm [hunk 7]
```diff
@@ -193,16 +164,76 @@
...
+	if(locate(/obj/item/soap/syndie) in items_list) //If we have syndie soap we have the other items
+		return
+	to_chat(usr, "<span class='notice'>You unlock [src]'s integrated deluxe cleaning supplies!</span>")
+	items_list += new /obj/item/soap/syndie(src) //We add not replace.
+	items_list += new /obj/item/reagent_containers/spray/cyborg_lube(src)
+	return TRUE
+
+/obj/item/organ/cyberimp/arm/service
+	name = "service toolset implant"
+	desc = "A set of tools needed for any barkeep or cook on the user's arm."
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]:
```suggestion
	desc = "A set of miscellaneous gadgets hidden behind a concealed panel on the user's arm."
```

Albeit this one implant is pretty much filler content and I honestly believe it's quite cumbersome to use.

File: code/modules/surgery/organs/augments_arms.dm [hunk 8]
```diff
@@ -231,7 +262,26 @@
...
+		return
+	to_chat(usr, "<span class='notice'>You unlock [src]'s integrated knife!</span>")
+	items_list += new /obj/item/assembly/flash/armimplant(src)
+	return TRUE
+
+/obj/item/organ/cyberimp/arm/shield/Initialize()
+	. = ..()
+	if(locate(/obj/item/assembly/flash/armimplant) in items_list)
+		var/obj/item/assembly/flash/armimplant/F = locate(/obj/item/assembly/flash/armimplant) in items_list
+		F.I = src
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: That is the baddest copypasta of this entire PR. Replace with the following.
```
	var/obj/item/assembly/flash/armimplant/F = new(src)
	F.I = src
	items_list += F
	return TRUE
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
The service implant is bad.

File: code/modules/surgery/organs/augments_arms.dm (hunk 8 above)
Comment (Trilbyspaceclone) [reply code/modules/surgery/organs/augments_arms.dm]: I did it correctly right?

File: code/modules/surgery/organs/augments_arms.dm (hunk 2 above)
Comment (Trilbyspaceclone) [reply code/modules/surgery/organs/augments_arms.dm]: Like this correct?

File: code/game/objects/items/shields.dm (hunk 1 above)
Comment (Trilbyspaceclone) [reply code/game/objects/items/shields.dm:254]: No longer a issue

File: code/modules/surgery/organs/augments_arms.dm [hunk 9]
```diff
@@ -158,7 +157,7 @@
 
 /obj/item/organ/cyberimp/arm/toolset/emag_act()
 	. = ..()
-	if(locate(/obj/item/kitchen/knife/combat/cyborg) in items_list)
+	if(obj_flags |= EMAGGED)
 		return
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]:
```
	if(obj_flags & EMAGGED)
		return
	obj_flags |= EMAGGED
```

File: code/modules/surgery/organs/augments_arms.dm [hunk 10]
```diff
@@ -171,20 +170,20 @@
 
 /obj/item/organ/cyberimp/arm/surgery/emag_act()
 	. = ..()
-	if(locate(/obj/item/kitchen/knife/combat/cyborg) in items_list)
+	if(obj_flags |= EMAGGED)
 		return
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: Idem.

File: code/modules/surgery/organs/augments_arms.dm [hunk 11]
```diff
@@ -171,20 +170,20 @@
...
 	to_chat(usr, "<span class='notice'>You unlock [src]'s integrated knife!</span>")
 	items_list += new /obj/item/kitchen/knife/combat/cyborg(src)
 	return TRUE
 
 /obj/item/organ/cyberimp/arm/janitor
-	name = "janitor toolset implant"
+	name = "janitorial tools implant"
 	desc = "A set of surgical tools hidden behind a concealed panel on the user's arm."
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: The description was the line I pointed out, not the name. It still says "surgical tools" instead of "janitorial tools".

File: code/modules/surgery/organs/augments_arms.dm [hunk 12]
```diff
@@ -193,12 +192,12 @@
...
 	name = "service toolset implant"
-	desc = "A set of tools needed for any barkeep or cook on the user's arm."
+	desc = "A set of miscellaneous gadgets hidden behind a concealed panel on the user's arm."
 	contents = newlist(/obj/item/extinguisher/mini, /obj/item/kitchen/knife/combat/bone/plastic, /obj/item/hand_labeler, /obj/item/pen, /obj/item/reagent_containers/dropper, /obj/item/kitchen/rollingpin, /obj/item/reagent_containers/glass/beaker/large, /obj/item/reagent_containers/syringe,/obj/item/reagent_containers/food/drinks/shaker, /obj/item/radio/off, /obj/item/camera, /obj/item/modular_computer/tablet/preset/cargo)
 
 /obj/item/organ/cyberimp/arm/service/emag_act()
 	. = ..()
-	if(locate(/obj/item/kitchen/knife/combat/cyborg) in items_list)
+	if(obj_flags |= EMAGGED)
 		return
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: Same as the first comment of this review.

File: code/modules/surgery/organs/augments_arms.dm [hunk 13]
```diff
@@ -243,8 +242,10 @@
 /obj/item/organ/cyberimp/arm/flash/Initialize()
 	. = ..()
 	if(locate(/obj/item/assembly/flash/armimplant) in items_list)
-		var/obj/item/assembly/flash/armimplant/F = locate(/obj/item/assembly/flash/armimplant) in items_list
+		var/obj/item/assembly/flash/armimplant/F = new(src)
 		F.I = src
+		items_list += F
+		return TRUE
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: No, I only told you to modify the riot shield emag_act , not the `Initialize` procedure of every `cyberimp/arm` that start out with a flash. Undo this change.

File: code/modules/surgery/organs/augments_arms.dm [hunk 14]
```diff
@@ -259,8 +260,10 @@
 /obj/item/organ/cyberimp/arm/combat/Initialize()
 	. = ..()
 	if(locate(/obj/item/assembly/flash/armimplant) in items_list)
-		var/obj/item/assembly/flash/armimplant/F = locate(/obj/item/assembly/flash/armimplant) in items_list
+		var/obj/item/assembly/flash/armimplant/F = new(src)
 		F.I = src
+		items_list += F
+		return TRUE
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: Same here.

File: code/modules/surgery/organs/augments_arms.dm [hunk 15]
```diff
@@ -274,14 +277,16 @@
 
 /obj/item/organ/cyberimp/arm/shield/emag_act()
 	. = ..()
-	if(locate(/obj/item/assembly/flash/armimplant) in items_list)
+	if(obj_flags |= EMAGGED)
 		return
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: Same as the first comment of this review.

File: code/modules/surgery/organs/augments_arms.dm [hunk 16]
```diff
@@ -274,14 +277,16 @@
...
-	to_chat(usr, "<span class='notice'>You unlock [src]'s integrated knife!</span>")
+	to_chat(usr, "<span class='notice'>You unlock [src]'s high-power flash!</span>")
 	items_list += new /obj/item/assembly/flash/armimplant(src)
 	return TRUE
 
 /obj/item/organ/cyberimp/arm/shield/Initialize()
 	. = ..()
 	if(locate(/obj/item/assembly/flash/armimplant) in items_list)
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: Remove these lines.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/game/objects/items/kitchen.dm [hunk 17]
```diff
@@ -138,6 +138,14 @@
 	throwforce = 15
 	materials = list()
 
+/obj/item/kitchen/knife/combat/bone/plastic
```
Comment (Ghommie) [inline code/game/objects/items/kitchen.dm:141]: This is an useless joke that perhaps should be removed.

File: code/modules/surgery/organs/augments_arms.dm [hunk 18]
```diff
@@ -231,7 +265,21 @@
...
+	desc = "A deployable riot shield to help deal with civil unrest."
+	contents = newlist(/obj/item/shield/riot/implant)
+
+/obj/item/organ/cyberimp/arm/shield/emag_act()
+	. = ..()
+	if(obj_flags & EMAGGED)
+		return
+	obj_flags |= EMAGGED
+	to_chat(usr, "<span class='notice'>You unlock [src]'s high-power flash!</span>")
+	items_list += new /obj/item/assembly/flash/armimplant(src)
```
Comment (Ghommie) [inline code/modules/surgery/organs/augments_arms.dm]: How many times do I have to tell you? Change this row with these three.
```
	var/obj/item/assembly/flash/armimplant/F = new(src)
	items_list += F
	F.I = src
```

File: code/game/objects/items/shields.dm [hunk 19]
```diff
@@ -237,3 +237,20 @@
...
 	transparent = FALSE
+
+/obj/item/shield/riot/implant
+	name = "riot tower shield"
+	desc = "A massive shield that can block a lot of attacks, can take a lot of abuse before braking."
+	item_state = "metal"
+	icon_state = "metal"
+	icon = 'icons/obj/items_and_weapons.dmi'
+	block_chance = 30 //May be big but hard to move around to block.
+	force = 10
```
Comment (Ghommie) [inline code/game/objects/items/shields.dm]: The parent type already has 10 force. Remove this line.

File: code/game/objects/items/shields.dm [hunk 20]
```diff
@@ -237,3 +237,20 @@
 	w_class = WEIGHT_CLASS_HUGE
 	item_flags = SLOWS_WHILE_IN_HAND
 	transparent = FALSE
+
+/obj/item/shield/riot/implant
+	name = "riot tower shield"
+	desc = "A massive shield that can block a lot of attacks, can take a lot of abuse before braking."
```
Comment (Ghommie) [inline code/game/objects/items/shields.dm]:
```suggestion
	desc = "A massive shield capable of substaining a lot of abuse before breaking."
```
And please learn the difference between break and brake for once.

Comment (Detective-Google) [inline code/game/objects/items/shields.dm]: `desc = "A massive shield that can block a lot of attacks, can take a lot of abuse before braking."`
*Breaking

**Review by Ghommie — CHANGES_REQUESTED** [review]
A couple suggestions I left pending days ago.

File: code/modules/surgery/organs/augments_arms.dm (hunk 18 above)
Comment (Trilbyspaceclone) [reply code/modules/surgery/organs/augments_arms.dm]: im so confused now.
You told me to revert that one???

File: code/game/objects/items/shields.dm [hunk 21]
```diff
@@ -237,3 +237,19 @@
 	w_class = WEIGHT_CLASS_HUGE
 	item_flags = SLOWS_WHILE_IN_HAND
 	transparent = FALSE
+
+/obj/item/shield/riot/implant
+	name = "riot tower shield"
+	desc = "A massive shield that can block a lot of attacks, can take a lot of abuse before braking." //It cant brake unless you somehow get it outside the implant
```
Comment (Arturlang) [inline code/game/objects/items/shields.dm]:
```suggestion
	desc = "A massive shield that can block a lot of attacks and can take a lot of abuse before breaking." //It cant break unless it is removed from the implant
```

**Review by Ghommie — APPROVED** [review]
